src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -623,6 +623,17 @@ private void OnSuggestedActionsChanged(Workspace currentWorkspace, DocumentId? c
                 }
 
                 var provider = _owner;
+
+                // The _owner flag is cleared on Dispose. This should only occur when cancellation has already been
+                // requested, but at least in 16.8 Preview 3.0 it was not always the case.
+                cancellationToken.ThrowIfCancellationRequested();
+                //Assumes.Present(provider);
```

Comment:
@AmadeusW The suggested actions source is getting disposed without first cancelling in-flight calls to `GetSuggestedActionCategoriesAsync`. This seems like a bug.

Reply:
@sharwell Can we file and link to a bug here?